commands/logs.go

```diff
@@ -28,6 +31,12 @@ func init() {
 }
 
 func logsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	if cmd.Flags().Changed("format") {
+		return errors.New("the \"--format\" flag cannot be used with this command")
+	} else if viper.GetString("format") == printer.FormatJSON {
+		return errors.New("json formatting cannot be applied on the output of this command")
```

Comment:
I'm happy making this distinction, good thinking! My only concern is that the error message doesn't explicitly point to the environment variable as it is now, and as sometimes those can be somehow obscure, I'd only suggest to change this to state that the value comes from the environment variable:

```suggestion
		return errors.New("json formatting cannot be applied on this command. Please check the value of \"MMCTL_FORMAT\"")
```

Reply:
Fair point, fixed!

```
$ MMCTL_FORMAT=json ./mmctl logs                                                              
Error: json formatting cannot be applied on this command. Please check the value of "MMCTL_FORMAT"
...
```